### PR TITLE
 Partial Fixes to #351, #352 as per 09/25/2018 TAPI call agreements

### PR DESCRIPTION
**Pull request by karthik-sethuraman**
- Renamed CentralFrequencyOrWavelength data-type to CentralFrequency
- Renamed Spectrum data-type to SpectrumBand
- Removed support for legacy definitions (m,n) of channel-number and
frequency-slot to compute CentralFrequency & SpectrumBand
- Refactored the frequency constraints properties of CentralFrequency
(grid-type and adjustment-granularity) into a separate (new) data-type
FrequencyConstraint.
- Added FrequencyConstraint property to CentralFrequency and
SpectrumBand data-types
- Renamed SpectrumType enumeration to PhotonicLayerQualifier
- Added 2 additional ModulationTechniques: PAM4 & PAM8
- Clarified comments/descriptions for threshold/power configuration
attributes

### CODE REVIEW

File: YANG/tapi-photonic-media@2018-08-31.yang [hunk 1]
```diff
@@ -508,27 +502,18 @@ module tapi-photonic-media {
...
-                    193.1 + <channelNumber> × <adjustmentGranularity> where channelNumber is a positive or negative integer including 0 and adjustment_granularity is the nominal central frequency granularity in THz";
-            }
-            description "This data-type holds the central frequency directly or optionally the information to determine the nominal central frequency of a FIXED grid (DWDM or CWDM) and FLEX grid type systems.
+            description "This data-type holds the central frequency information as well frequency constraints in terms of GridType ( FIXED grid (DWDM or CWDM) or FLEX grid) and AdjustmentGranularity.
                 As per ITU-T G.694.1, the nominal central frequency (in THz) is calculated as follows:
-                193.1 + <channelNumber> × <adjustmentGranularity> where channelNumber is a positive or negative integer including 0 and <adjustment_granularity> is the nominal central frequency granularity in THz
+                193.1 + <channelNumber> × <adjustmentGranularity> where channelNumber is a positive or negative integer including 0 and <adjustment_granularity> is the channel spacing in THz
```
Comment (arthurMll) [inline YANG/tapi-photonic-media@2018-08-31.yang]: I think this definition is not longer valid as channel spacing is no longer included in the model.

File: YANG/tapi-photonic-media@2018-08-31.yang [hunk 2]
```diff
@@ -742,5 +716,20 @@ module tapi-photonic-media {
...
+                type adjustment-granularity;
+                description "Adjustment granularity in Gigahertz. As per ITU-T G.694.1, it is used to calculate nominal central frequency (in THz)";
+            }
+            leaf grid-type {
+                type grid-type;
+                description "Specifies the frequency grid standard used to determine the nominal central frequency and frequency slot width";
+            }
+            description "This data-type holds the frequency constraint information in terms of GridType ( FIXED grid (DWDM or CWDM) or FLEX grid) and AdjustmentGranularity.
+                As per ITU-T G.694.1, the nominal central frequency (in THz) is calculated as follows:
+                193.1 + <channelNumber> × <adjustmentGranularity> where channelNumber is a positive or negative integer including 0 and <adjustment_granularity> is the channel spacing in THz
```
Comment (arthurMll) [inline YANG/tapi-photonic-media@2018-08-31.yang]: I think this definition is not longer valid as channel spacing is no longer included in the model.

File: YANG/tapi-photonic-media@2018-08-31.yang [hunk 3]
```diff
@@ -250,15 +250,11 @@ module tapi-photonic-media {
                 uses otsi-termination-config-pac;
                 description "none";
             }
-            container nmc-config {
```
Comment (arthurMll) [inline YANG/tapi-photonic-media@2018-08-31.yang]: I am not sure if this container is not needed anymore. I wonder how you will specify the media-channel spectrum band if it is not througth this configuration.

File: YANG/tapi-photonic-media@2018-08-31.yang (hunk 1 above)
Comment (karthik-sethuraman) [reply YANG/tapi-photonic-media@2018-08-31.yang]: Agree - fixed the description as per commit below

File: YANG/tapi-photonic-media@2018-08-31.yang (hunk 2 above)
Comment (karthik-sethuraman) [reply YANG/tapi-photonic-media@2018-08-31.yang]: Agree - fixed the description as per commit below

File: YANG/tapi-photonic-media@2018-08-31.yang (hunk 3 above)
Comment (karthik-sethuraman) [reply YANG/tapi-photonic-media@2018-08-31.yang]: Agree - added the spectrum property to otsi-termination-config-pac as per commit below

**Pull request merged**